embedded/remotestorage/s3/s3.go

```diff
@@ -41,17 +43,22 @@ import (
 )
 
 type Storage struct {
-	endpoint    string
-	accessKeyID string
-	secretKey   string
-	bucket      string
-	prefix      string
-	location    string
-	httpClient  *http.Client
+	endpoint                   string
+	S3RoleEnabled              bool
+	s3Role                     string
+	accessKeyID                string
+	secretKey                  string
+	bucket                     string
+	prefix                     string
+	location                   string
+	httpClient                 *http.Client
+	s3CredentialsRefreshTicker *time.Ticker
```

Comment:
`s3CredentialsRefreshTicker *time.Ticker` seems to be used in the scope of the method, it should not be part of the struct...

Reply:
ah, it's for a long-running goroutine,  probably it's preferable to have the reference to the goroutine directly.

The goroutine should not hardly stop if it fails to fetch to renew the credentials but it may log the error and keep trying